tests/subgraph/mapping-v3-core/mapping-v3-engine-flex-core.test.ts

```diff
@@ -0,0 +1,975 @@
+  TRANSFER_ENTITY_TYPE,
+  ONE_MILLION,
+  booleanToString,
+  TEST_CONTRACT,
+  TEST_SUPER_ADMIN_ADDRESS,
+  WHITELISTING_ENTITY_TYPE,
+  IPFS_CID,
+  PROJECT_EXTERNAL_ASSET_DEPENDENCY_ENTITY_TYPE,
+  IPFS_CID2,
+} from "../shared-helpers";
+import {
+  mockProjectScriptDetailsCall,
+  mockProjectStateDataCall,
+  testProjectDetailsUpdated,
+  testProjectScriptDetailsUpdated,
+  testProjectStateDataUpdated,
+  mockRefreshContractCalls
+} from "./helpers";
+import {
+  Project,
+  Contract,
+  ProjectExternalAssetDependency
+} from "../../../generated/schema";
+import {
+  Mint,
+  ProjectUpdated,
+  PlatformUpdated
+} from "../../../generated/IGenArt721CoreV3_Base/IGenArt721CoreContractV3_Base";
+import { Transfer } from "../../../generated/IERC721GenArt721CoreV3Contract/IERC721";
+import { OwnershipTransferred } from "../../../generated/OwnableGenArt721CoreV3Contract/Ownable";
+import { SuperAdminTransferred } from "../../../generated/AdminACLV0/IAdminACLV0";
+import {
+  ExternalAssetDependencyUpdated,
+  ExternalAssetDependencyRemoved,
+  GatewayUpdated,
+  ProjectExternalAssetDependenciesLocked
+} from "../../../generated/IGenArt721CoreV3_Engine_Flex/IGenArt721CoreContractV3_Engine_Flex";
+import {
+  FIELD_PROJECT_ACTIVE,
+  FIELD_PROJECT_ARTIST_ADDRESS,
+  FIELD_PROJECT_ARTIST_NAME,
+  FIELD_PROJECT_ASPECT_RATIO,
+  FIELD_PROJECT_BASE_URI,
+  FIELD_PROJECT_COMPLETED,
+  FIELD_PROJECT_CREATED,
+  FIELD_PROJECT_DESCRIPTION,
+  FIELD_PROJECT_LICENSE,
+  FIELD_PROJECT_MAX_INVOCATIONS,
+  FIELD_PROJECT_NAME,
+  FIELD_PROJECT_PAUSED,
+  FIELD_PROJECT_SCRIPT,
+  FIELD_PROJECT_SCRIPT_TYPE,
+  FIELD_PROJECT_SECONDARY_MARKET_ROYALTY_PERCENTAGE,
+  FIELD_PROJECT_WEBSITE,
+  handleIAdminACLV0SuperAdminTransferred,
+  handleMint,
+  handleTransfer,
+  handlePlatformUpdated,
+  handleOwnershipTransferred,
+  handleProjectUpdated,
+  handleExternalAssetDependencyUpdated,
+  handleExternalAssetDependencyRemoved,
+  handleGatewayUpdated,
+  handleProjectExternalAssetDependenciesLocked
+} from "../../../src/mapping-v3-core";
+import {
+  generateContractSpecificId,
+  generateProjectScriptId,
+  generateWhitelistingId
+} from "../../../src/helpers";
+import { FLEX_CONTRACT_EXTERNAL_ASSET_DEP_TYPES } from "../../../src/constants";
+
+const randomAddressGenerator = new RandomAddressGenerator();
+
+const coreType = "GenArt721CoreV3_Engine_Flex";
+
+test(`${coreType} Can add/update a project external asset dependency`, () => {
+  clearStore();
+  // Add project to store
+  const projectId = BigInt.fromI32(0);
+  mockCoreType(TEST_CONTRACT_ADDRESS, `${coreType}`);
+  const fullProjectId = generateContractSpecificId(
+    TEST_CONTRACT_ADDRESS,
+    projectId
+  );
+  addNewProjectToStore(
+    TEST_CONTRACT_ADDRESS,
+    projectId,
+    "Test Project",
+    randomAddressGenerator.generateRandomAddress(),
+    BigInt.fromI64(i64(1e18)),
+    CURRENT_BLOCK_TIMESTAMP.minus(BigInt.fromI32(100))
+  );
+
+  const event: ExternalAssetDependencyUpdated = changetype<
```

Comment:
Super minor nit but would be helpful when reading this test if this event was called updateEvent1 or creationEvent given the context of the other 2 events